index.bs

```diff
@@ -3199,6 +3464,177 @@ The [=remote end steps=] with |command parameters| are:
 1. Return a new map matching the <code>ScriptEvaluateResultSuccess</code>
    production, with the <code>result</code> field set to |result|.
 
+#### The script.callFunction Command ####  {#command-script-callFunction}
+
+Issue: Add `this` parameter.
+
+The <dfn export for=commands>script.callFunction</dfn> command calls a provided
+function with given arguments in a given realm.
+
+ <code>RealmInfo</code> can be either a realm or a browsing context.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+    <pre class="cddl remote-cddl">
+      ScriptCallFunctionCommand = {
+        method: "script.callFunction",
+        params: ScriptCallFunctionParameters
+      }
+
+      ScriptCallFunctionParameters = {
+        functionDeclaration: text;
+        arguments: [ArgumentValue];
+        ?awaitPromise: bool;
+        target: Target;
+      }
+
+      ArgumentValue = (
+        RemoteReference //
+        LocalValue
+      );
+
+    </pre>
+   </dd>
+   <dt>Return Type</dt>
+   <dd>
+    <pre class="cddl local-cddl">
+      ScriptCallFunctionResult = (
+        ScriptCallFunctionResultSuccess //
+        ScriptCallFunctionResultException
+      )
+
+      ScriptCallFunctionResultSuccess = {
+         result: RemoteValue
+      }
+
+      ScriptCallFunctionResultException = {
+        exceptionDetails: ExceptionDetails
+      }
+    </pre>
+   </dd>
+</dl>
+
+Issue: TODO: Add timeout argument as described in the script.evaluate.
+
+<div algorithm>
+To <dfn>deserialize an argument value</dfn> given an |serialized argument|:
+
+1. If |serialized argument| is [=RemoteReference=], return [=deserialize a remote reference=] of
+   |serialized argument|.
+
+1. Return [=deserialize a local value=] of |serialized argument|.
+
+</div>
+
+<div algorithm>
+
+To <dfn>deserialize arguments</dfn> given |serialized arguments list|:
+
+1. Let |deserialized arguments list| be an empty list.
+
+1. For each |serialized argument| of |serialized arguments list|:
+
+  1. Let |deserialized argument| be the result of [=try=] to [=deserialize an argument value=] with
+     given |serialized argument|.
+
+  1. If |deserialized argument| is [=error=], return |deserialized argument|.
+
+  1. Append |deserialized argument| to the |deserialized arguments list|.
+
+1. Return [=success=] with data |deserialized arguments list|.
+
+Issue: Clarify behaviour in case of deserialization failed.
+</div>
+
+<div algorithm>
+To <dfn>evaluate function body</dfn> with a given |function declaration|,
+|environment settings|, |base URL|, and |options|:
+
+Note: the |function declaration| is parenthesized and evaluated.
+
+1. Let |function script| be the result of [=create a classic script=] with
+   <code>(</code> + |function declaration| + <code>)</code>, |environment settings|, |base URL|, and |options|.
```

Comment:
I think you need to use [string concatenate](https://infra.spec.whatwg.org/#string-concatenate) from the infra spec here, something like `Let |body| be [=concatenate=] «"<code>(</code>", |function declaration|, "<code>)</code>"»`.

Reply:
done, thanks!